test/parallel/test-vm-module-referrer-realm.js

```diff
@@ -0,0 +1,72 @@
+// Flags: --experimental-vm-modules
+'use strict';
+const common = require('../common');
+
+const assert = require('assert');
+const { Script, SourceTextModule, createContext } = require('vm');
+
+async function test() {
+  const foo = new SourceTextModule('export const a = 1;');
+  await foo.link(common.mustNotCall());
+  await foo.evaluate();
+
+  const ctx = createContext({}, {
+    importModuleDynamically: common.mustCall((specifier, wrap) => {
+      assert.strictEqual(specifier, 'foo');
+      assert.strictEqual(wrap, ctx);
+      return foo;
+    }, 2),
+  });
+  {
+    const s = new Script('Promise.resolve("import(\'foo\')").then(eval)', {
+      importModuleDynamically: common.mustNotCall(),
+    });
+
+    const result = s.runInContext(ctx);
+    assert.strictEqual(foo.namespace, await result);
```

Comment:
Isn't that backward? IIUC the expected value is `foo.namespace`, the actual value is `await result`
```suggestion
    assert.strictEqual(await result, foo.namespace);
```

Reply:
Right, fixed! Thank you.